graphrag/index/run/workflow.py

```diff
@@ -48,8 +48,13 @@ async def _emit_workflow_output(
 ) -> pd.DataFrame:
     """Emit the workflow output."""
     output = cast(pd.DataFrame, workflow.output())
-    for emitter in emitters:
-        await emitter.emit(workflow.name, output)
+    rows, cols = output.shape
+    # only write the final output if it has content
+    # this is expressly designed to allow us to create
+    # workflows with side effects that don't produce a formal output to save
+    if rows > 0 and cols > 0:
```

Comment:
What about `if not output.empty`? The [docs](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.empty.html) suggest this better accounts for the edge case where there may be columns but no rows in a dataframe.

Reply:
This is why we have better pandas users than myself review!